src/waveform/renderers/waveformmarkset.cpp

```diff
@@ -80,7 +80,7 @@ void WaveformMarkSet::setup(const QString& group, const QDomNode& node,
 }
 
 void WaveformMarkSet::clear() {
-    m_defaultMark.reset();
+    m_defaultMark.WaveformMark::~WaveformMark();
```

Comment:
The destructor is called automatically if m_defaultMark falls out of scope. In this case when WaveformMarkSet is deleted. 

The intended change here is to make m_defaultMark a pointer 
WaveformMark* m_pDefaultMark; 

Instead of setup() we can allocate it on the heap: 
m_pDefaultMark = new WaveformMark( ... 

and here we can call delete m_pDefaultMark; 

It is a good practice to avoid plain pointers, so you may replace them by unique_ptr<WaveformMark> in a next step.